manifests/repository.pp

```diff
@@ -11,7 +11,7 @@
   $url = "http://download.bareos.org/bareos/release/${release}/"
 
   $os = $::operatingsystem
-  $osrelease = $::operatingsystemrelease
+  $osrelease = $::os[release][major]
```

Comment:
this should fix the error and adds indirect support for old and new puppet release `$osrelease = pick($facts['os']['release']['major'], $::operatingsystemrelease)`

Reply:
works with centos 7